### PR TITLE
Implement admin page for gift cards

### PR DESCRIPTION
**Pull request by BerglundDaniel**
This PR adds an admin page for the gift cards with related entities etc. The view is similar to the one for transactions.

Screenshot from 2023-12-04 15-18-41

Screenshot from 2023-12-04 15-18-22

## Summary by CodeRabbit

- **New Features**
  - Introduced Gift Card management with the ability to view and delete gift cards.
  - Added a new Gift Card List component to display all gift cards with details.
  - Created a Gift Card Show component to present individual gift card information and associated products.
  - Incorporated new navigation menu entry for Gift Cards in the admin panel.

- **Enhancements**
  - Updated Order List with improved rendering and date-time display.

- **API & Database Updates**
  - Implemented new routes and entities for handling gift cards.
  - Adjusted database schema for gift cards, including status and amount fields.

- **Bug Fixes**
  - Fixed issues with initial data creation for shop transactions and gift cards.

### CODE REVIEW

File: api/src/firstrun.py [hunk 1]
```diff
@@ -411,17 +411,27 @@ def create_shop_accounts_cost_centers() -> None:
...
         name="Makerspace access starter pack",
     )
 
-    get_or_create(ProductGiftCardMapping, gift_card_id=gift_card.id, product_id=product.id)
+    gift_card = get_or_create(
+        GiftCard,
+        amount=product.price,
+        validation_code=12989519,
+        email="test@fake.com",
+        status="activated",
```
Comment (emanuelen5) [inline api/src/firstrun.py:425]: What does the `activated` status mean for a gift card? Has that been used?

I would think a gift card had the following states:
- `Valid` (when it has been bought and not yet used)
- `Used` (when it has been used by a customer and cannot be used again)
- `Cancelled` (if we have paid it back to the customer for whatever reason, and the customer should not be allowed to use it)

**Comment by emanuelen5** [comment]
I think it would be nice if the admin page had a way to cancel the gift cards in case they are paid back. This can be implemented in a later PR (just create an issue for it, in that case).

File: api/src/firstrun.py (hunk 1 above)
Comment (BerglundDaniel) [reply api/src/firstrun.py:425]: Cancelling would be useful, we should add that

**Comment by BerglundDaniel** [comment]
I opened a new issue about cancelling gift cards

**Comment by BerglundDaniel** [comment]
@coderabbitai review

**Comment by BerglundDaniel** [comment]
@coderabbitai review

**Pull request merged**